lib/block_view.go

```diff
@@ -6881,6 +6956,22 @@ func (bav *UtxoView) _connectUpdateNFT(
 			txMeta.NFTPostHash.String())
 	}
 
+	if isBuyNow && blockHeight < BuyNowNFTBlockHeight {
```

Comment:
The current way the code is written, we accept any "IsBuyNowKey" as being legit even if we have not passed the BuyNowNFTBlockHeight.  This means that someone could see this PR and add the IsBuyNowKey to extra data to a transaction right now in order to break nodes that sync from scratch when this update is merged in the future. Instead, it seems better to wrap the code above that checks for the IsBuyNowKey in extra data in a blockheight check.  That way, "isBuyNow" is forced to be false until the BuyNowNFTBlockHeight hits.